src/plugins/intel_gpu/src/graph/broadcast.cpp

```diff
@@ -82,8 +82,12 @@ std::vector<layout> broadcast_inst::calc_output_layouts(broadcast_node const& /*
                                                      static_cast<void*>(target_shape.data()));
         const_data.emplace(1, target_shape_tensor);
         ov::op::v3::shape_infer(&op, input_shapes, output_shapes, const_data);
+    } else if (impl_param.input_layouts.size() >= 2 && impl_param.get_input_layout(1).is_static()) {
+        // target shape rank is set as second input.
+        auto output_rank = impl_param.get_input_layout(1).get_dim(0);
+        output_shapes[0] = ShapeType::dynamic(std::max(output_rank, static_cast<int>(1)));
```

Comment:
Cant we merge this block and the next condition block with removing impl_param.get_input_layout(1).is_static() ?

Reply:
Done